fix(runtime): repeated gO in markdown adds extra toc indentation

tanloong opened the pull request:
Fix #33185

tanloong committed:
fix(runtime): repeated gO in markdown adds extra toc indentation

clason commented:
Thank you, but I'm already working on a different fix.

tanloong commented:
> Thank you, but I'm already working on a different fix.

OK. Notice your comment at #33185 just after creating this PR. Been excited about my first contrib to neovim, for a moment.

clason commented:
Now I feel bad...

My fix is to set the indents in the memoized function (modify `text` before inserting it in the table); if you want to do that, go ahead!

clason commented:
Alternative would be just to remove the memorization; not sure how much difference that makes (in a large markdown file)?

tanloong commented:
Thank you very much, really appreciate it! Reverted my change and modified the memoized function.

I think keeping the memorization is safer? Current fix causes less performance decline than removing the memorization will do?

clason commented:
It could (hence my question), but I added it without deeper thought, so since it did lead to problems, it might be better to remove it until someone actually complains about performance issues.
